compiler/src/iree/compiler/Codegen/ExternalInterfaces/Utils.h

```diff
@@ -16,6 +18,34 @@ namespace mlir::iree_compiler::IREE {
 
 static const char kEncodingInfoAttrName[] = "encoding_info";
 
+template <typename DeviceEncodingLayoutResolverAttrInterface,
+          typename EncodingLayoutAttr>
+struct WrappedExternalModel
```

Comment:
nit: I think `DeviceEncodingLayoutResolverExternalModelBase` is a better name. Also, can you add a doc to the class?